src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Colors = 'red' | 'green' | 'blue';
+type Shapes = `triangle` | `circle` | `rectangle`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shapes;
+  color: Colors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shapes = `triangle`;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid triangle dimensions');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle dimensions');
+    }
+  }
+
+  getArea = (): number => {
+    const { a, b, c } = this;
+
+    const area = (a + b + c) / 2;
+
+    return +Math.sqrt(area * (area - a) * (area - b) * (area - c)).toFixed(2);
+  };
+}
+
+export class Circle implements Figure {
+  readonly shape: Shapes = `circle`;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Colors,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+  }
+
+  getArea = (): number => {
+    const { radius } = this;
+    const area = Math.PI * radius * radius;
+
+    return Math.floor(area * 100) / 100;
+  };
+}
+
+export class Rectangle implements Figure {
+  readonly shape: Shapes = `rectangle`;
+
+  constructor(
+    public color: Colors,
+    public width: number,
+    public height: number,
```

Comment:
[CODE KNOWLEDGE] - The properties 'color', 'width', and 'height' do not need to be manually declared in the class body since TypeScript will automatically create them when you use the 'public' keyword in the constructor.